src/physics/thermal_conduction.cpp

```diff
@@ -106,7 +119,25 @@ void ThermalConduction::completeSetup()
   if (timestepper_ != serac::TimestepMethod::QuasiStatic) {
     // If dynamic, assemble the mass matrix
     M_form_ = temperature_.createOnSpace<mfem::ParBilinearForm>();
-    M_form_->AddDomainIntegrator(new mfem::MassIntegrator());
+
+    // If both a specific heat capacity and density have been specified, use their product as the mass matrix
+    // coefficient
+    if (cp_ != nullptr && rho_ != nullptr) {
+      auto product_function = [](double a, double b) { return a * b; };
+      mass_coef_ = std::make_unique<TransformedScalarCoefficient>(std::move(rho_), std::move(cp_), product_function);
+    }
+    // otherwise, just use the one that is specified
+    else if (cp_ != nullptr) {
+      mass_coef_ = std::move(cp_);
+    } else if (rho_ != nullptr) {
```

Comment:
Minor: `std::unique_ptr`s have an implicit `operator bool()`, so this could just be
```cpp
if (cp_ && rho_)
```
etc - personally, I find it a little more readable that way.